### PR TITLE
Add regular execution integration tests

### PR DESCRIPTION
**Pull request by tnek**
All of the groundwork's there now for testing install -> sync -> running a binary

### CODE REVIEW

File: Testing/integration/SNTExecTest.m [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  XCTestExpectation *expectation =
+    [self expectationWithDescription:@"Wait for test binary to execute"];
+
+  dispatch_async(dispatch_get_global_queue(QOS_CLASS_BACKGROUND, 0), ^{
+    NSTask *task = [[NSTask alloc] init];
+    task.launchPath = path;
+    [task launch];
+    while ([task isRunning])
+      usleep(100000);  // 0.1s
```
Comment (pmarkowsky) [inline Testing/integration/SNTExecTest.m:39]:
```suggestion
    [task waitUntilExit];
```

Shouldn't it just be this? Then if your expectation isn't met within the alloted time you can just kill it with `[task terminate];`

File: Testing/integration/SNTExecTest.m [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    status = [task terminationStatus];
+    [expectation fulfill];
+  });
+
+  [self waitForExpectations:@[ expectation ] timeout:20.0];
```
Comment (pmarkowsky) [inline Testing/integration/SNTExecTest.m:44]: We can change this to `waitForExpectationsWithTimeout:handler:` to force terminate the task.

**Review by pmarkowsky — COMMENTED** [review]
Overall this looks good. I think we can get rid of the sleeps and make this cleaner.

File: Testing/integration/SNTExecTest.m (hunk 2 above)
Comment (tnek) [reply Testing/integration/SNTExecTest.m]: Done

File: Testing/integration/SNTExecTest.m (hunk 1 above)
Comment (tnek) [reply Testing/integration/SNTExecTest.m]: Done

**Review by pmarkowsky — APPROVED** [review]
LGTM.

**Pull request merged**